source/core/slang-string-util.cpp

```diff
@@ -0,0 +1,29 @@
+#include "slang-string-util.h"
+
+namespace Slang {
+
+/* static */void StringUtil::split(const UnownedStringSlice& in, char splitChar, List<UnownedStringSlice>& slicesOut)
```

Comment:
That's another option. I think of most methods of the base type to be the most commonly used and/or smallest functions. split could be in there. But there can be lots of ways to split (with substring) or one of a set of characters etc. 
I think we can get more work with less code out of the 'string-like functions' but that's probably another discussion.
I'd agrees the StringUtil class is a little silly right now cos it only has a single method. I think that it won't take long before it has a bunch of new friends tho.

Reply:
The other option (and this applies to the `RenderApiUtils` type too) is to just make those be global-scope functions and not scope them in any way.

I get the conventional wisdom that it is generally good in C++ to make as few things be member functions as possible, and to define everything else externally. The main benefit of member functions, though, is that they autocomplete easily when you do `someVal.` and get a list of everything it supports.

I'm not sold on functions that are neither in the global scope nor in the type they work on. It seems like introducing namespaces *without* the ability to get rid of the clutter they introduce via `using`.

Reply:
In large code bases I've worked with previously - the design philosopy was to have the minimum of functionality as methods of a type. Minimum - means in terms of typical usage. That it was good to push out other functionality into Util which are typically collections of related functions. 
This removes clutter and complexity from the type, reduces header size. It can be used as a mechanism to extend a type. 
```The main benefit of member functions, though, is that they autocomplete easily when you do someVal. and get a list of everything it supports.```
And with Util you get this - and are indeed a large part of the argument for using the mechanism. 
Free functions often end up being in practice not significantly more concise (and they are now polluting a much larger namespace, and can't be looked up) - because the Util name (and overriding comments) provide context - both in typename and in overall intent of collection of functions. 
The other advantage is the Util can also have functions that encapsulate functionality on multiple types - without any circular dependency issues.
The Util mechanism can be seen as a way of extending a functionality of a class and there may be lots of Utils that address different sets of functionality for a type. But it's more flexible than that - it's a mechanism to just group related functions together. 
Util also allows for grouping/scoping types - specific to the Util. RenderApiUtil does this with Info. Using this can add to conciseness.

So I would argue in most situations if a free floating function is part of a public interface is a sign of code smell. Free functions that are private to implementation are not a problem.